lib/keystore.cc

```diff
@@ -96,6 +100,50 @@ static rpmRC write_key_to_disk(rpmPubkey key, string & dir, string & filename, i
     return rc;
 }
 
+rpmRC rebuild_dir(rpmtxn txn, keystore * store, rpmKeyring keys)
+{
+
+    auto macros = rpm::macros();
+    auto [mrc, keyringpath] = macros.expand("%{_keyringpath}");
+    while ('/' == keyringpath.back())
+	keyringpath.pop_back();
+    std::string path = rpm::expand_path({rpmtxnRootDir(txn), keyringpath});
+    std::string tmppath = keyringpath + ".tmp/";
+    std::string oldpath = keyringpath + ".rpmold/";
```

Comment:
Include the rpmtxnRootDir() in these already, as it is you just need to duplicate that work multiple times in the below, making it harder to follow.